packages/main/src/TimePicker.js

```diff
@@ -0,0 +1,640 @@
+
+		array = array.map(i => i.toString());
+
+		return array;
+	}
+
+	get secondsArray() {
+		return this.generateTimeItemsArray(60);
+	}
+
+	get minutesArray() {
+		return this.generateTimeItemsArray(60);
+	}
+
+	get hoursArray() {
+		let hoursValueArray = [];
+
+		if (this._hoursParameters.isTwelveHoursFormat) {
+			hoursValueArray = this.generateTimeItemsArray(12);
+		} else {
+			hoursValueArray = this.generateTimeItemsArray(24);
+		}
+
+		if (this._hoursParameters.minHour === 1) {
+			hoursValueArray = hoursValueArray.map(x => (x * 1 + 1).toString());
+		}
+
+		return hoursValueArray;
+	}
+
+	get periodsArray() {
+		return this.getFormat().aDayPeriods.map(x => x.toUpperCase());
+	}
+
+	_getInput() {
+		return this.shadowRoot.querySelector("ui5-input");
+	}
+
+	submitPickers() {
+		const selectedDate = new Date(),
+			secondsSlider = this.getStaticAreaItemDomRef().querySelector(".ui5-timepicker-seconds-wheelslider"),
+			minutesSlider = this.getStaticAreaItemDomRef().querySelector(".ui5-timepicker-minutes-wheelslider"),
+			hoursSlider = this.getStaticAreaItemDomRef().querySelector(".ui5-timepicker-hours-wheelslider"),
+			periodsSlider = this.getStaticAreaItemDomRef().querySelector(".ui5-timepicker-period-wheelslider"),
+			hours = hoursSlider ? hoursSlider.getAttribute("value") : this._hoursParameters.minHour.toString(),
+			minutes = minutesSlider ? minutesSlider.getAttribute("value") : "0",
+			seconds = secondsSlider ? secondsSlider.getAttribute("value") : "0",
+			period = periodsSlider ? periodsSlider.getAttribute("value") : this.periodsArray[0];
+
+		if (period === this.periodsArray[1]) {
+			selectedDate.setHours(hours * 1 + 12);
+		} else {
+			selectedDate.setHours(hours);
+		}
+		selectedDate.setMinutes(minutes);
+		selectedDate.setSeconds(seconds);
+
+		this.setValue(this.getFormat().format(selectedDate));
+
+		this.fireEvent("change", { value: this.value, valid: true });
+
+		this.closePicker();
+	}
+
+	/**
+	 * Opens the picker.
+	 * @public
+	 */
+	isValid(value = "") {
+		return !!(value && this.getFormat().parse(value));
+	}
+
+	normalizeValue(sValue) {
+		return this.getFormat().format(this.getFormat().parse(sValue));
+	}
+
+	get _formatPattern() {
+		return this.formatPattern || "medium"; // get from config
+	}
+
+	get _isPattern() {
+		return this._formatPattern !== "medium" && this._formatPattern !== "short" && this._formatPattern !== "long";
+	}
+
+	get _displayFormat() {
+		return this.getFormat().oFormatOptions.pattern;
+	}
+
+	get _placeholder() {
+		return this.placeholder !== undefined ? this.placeholder : this._displayFormat;
+	}
+
+	handleSliderClicked(event) {
+		if (event.target._expanded) {
+			this.openSlider(event.target.label);
+		}
+	}
+
+	openSlider(label) {
+		for (let i = 0; i < this._slidersDomRefs.length; i++) {
+			if (this._slidersDomRefs[i].label !== label) {
+				this._slidersDomRefs[i].collapseSlider();
+			}
+		}
+	}
+
+	_onfocuscontainerin(e) {
+		if (e.target !== e.currentTarget) {
```

Comment:
The focus should cycle between the sliders and the buttons in the footer, at least this is how implemented in openui5 and I believe this is how it should be

Reply:
Done